explainability/core/counterfactuals/src/main/java/org/kie/kogito/explainability/local/counterfactual/CounterfactualExplainer.java

```diff
@@ -69,6 +70,24 @@ public CounterfactualExplainer(DataDomain dataDomain, List<Boolean> contraints,
         this.executor = executor;
     }
 
+    /**
+     * Create a new {@link CounterfactualExplainer} using OptaPlanner as the underlying engine.
+     *
+     * @param dataDistribution Characteristics of the data distribution as {@link DataDistribution}, if available
+     * @param dataDomain       A {@link DataDomain} which specifies the search space domain
+     * @param contraints       A list specifying by index which features are constrained
+     * @param goal             A collection of {@link Output} representing the desired outcome
+     * @param solverConfig     An OptaPlanner {@link SolverConfig} configuration
+     */
+    public CounterfactualExplainer(DataDistribution dataDistribution, DataDomain dataDomain, List<Boolean> contraints, List<Output> goal, SolverConfig solverConfig, Executor executor) {
```

Comment:
I think we already have something like 6 or more different `CounterfactualExplainer` constructors :)
What about create a nested `Builder` class to collect all params and then in the `build()` method invoke always the same constructor?
Something like [this](https://github.com/kiegroup/drools/blob/master/kie-pmml-trusty/kie-pmml-models/kie-pmml-models-drools/kie-pmml-models-drools-scorecard/kie-pmml-models-drools-scorecard-model/src/test/java/org/kie/pmml/models/drools/scorecard/model/KiePMMLScorecardModelTest.java#L92-L95)

Reply:
very good point :)
implemented.